src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -57,5 +58,8 @@ def contract_update_list_success(updated_resource, resource_client):
     _create_request, _created_model, _update_request, updated_model = updated_resource
     assert resource_client.is_primary_identifier_equal(
         resource_client.primary_identifier_paths, _created_model, updated_model
-    )
-    assert test_model_in_list(resource_client, updated_model)
+    ), "The primaryIdentifier returned in every progress event must match\
+         the primaryIdentifier passed into the request"
+    assert test_model_in_list(
+        resource_client, updated_model
+    ), "A list handler MUST always return a status of SUCCESS"
```

Comment:
This is checking that the updated model is returned by list so the assertion message should mention that. We don't require the list handler always return "SUCCESS". Just that it doesn't return IN_PROGRESS

Reply:
I would leave the success statement out. It does not require to be added here